api/envoy/config/route/v3/route_components.proto

```diff
@@ -1929,6 +1929,10 @@ message HeaderMatcher {
   // * The regex ``\d{3}`` does not match the value *1234*, so it will match when inverted.
   // * The range [-10,0) will match the value -1, so it will not match when inverted.
   bool invert_match = 8;
+
+  // If true, indicates the exact_match/prefix_match/suffix_match/contains_match should be case
+  // insensitive. This has no effect for the safe_regex_match.
```

Comment:
Why does this not apply to `safe_regex_match` as well?

Reply:
I'm following the same [ignore_case](https://github.com/envoyproxy/envoy/blob/dfef049fe9692ff8188a83bab859b310c212ece1/api/envoy/type/matcher/v3/string.proto#L65) in StringMatcher which also does not apply to regex, I think probably the reason is you can always enable the case-insensitive match easily with the `(?i)` flag directly in the regular expression itself?

Reply:
Oh, interesting -- I didn't realize that.  And it looks like gRPC C-core has the wrong behavior here.  I'll fix that separately.

Reply:
If you think it would be useful, I think we can support it in StringMatcher in a follow PR later.

Reply:
No, that's okay.  I've already fixed gRPC C-core to match what Envoy does (grpc/grpc#26567).  The other gRPC languages do the same thing.